### DISCUSSION TITLE
We need socks5 proxy option!

### DISCUSSION THREAD

**Discussion by btaskel** [community/community#141126] [Mobile]
Necessity: I come from a region where the internet is heavily censored, and access to the GitHub website is intermittently blocked. Therefore, I need to enable a VPN on my device. However, I do not want to enable the VPN solely for the GitHub app, **as it would increase the power consumption of mobile devices**. Using SOCKS5, on the other hand, would reduce power consumption and allow more flexible switching between proxy servers.

Technology: In the Android system, **WebView itself supports the configuration of a SOCKS5 proxy**. GitHub app developers can refer to the relevant documentation to obtain interface information.

**Comment by VeerammalV** [comment]
It sounds like you're looking for a solution to reduce power consumption while maintaining access to GitHub through a VPN. Since Android’s WebView supports SOCKS5 proxy configuration, you could suggest that GitHub app developers consider implementing SOCKS5 proxy support directly in the app. This would allow users to connect through proxy servers without needing to enable a full VPN, thereby saving power and offering more flexibility in proxy server switching.
